src/Refactoring-Core/RBFindAndReplaceRefactoring.class.st

```diff
@@ -0,0 +1,192 @@
+Class {
+	#name : #RBFindAndReplaceRefactoring,
+	#superclass : #RBMethodRefactoring,
+	#instVars : [
+		'method',
+		'selector',
+		'replacesAllHierarchy',
+		'matchNodes'
+	],
+	#category : #'Refactoring-Core-Refactorings'
+}
+
+{ #category : #'as yet unclassified' }
+RBFindAndReplaceRefactoring class >> find: aMethod of: aClass inAllHierarchy: aBoolean [
+	^ self new 
+		find: aMethod 
+		of: aClass
+		inAllHierarchy: aBoolean;
+		yourself
+]
+
+{ #category : #'as yet unclassified' }
+RBFindAndReplaceRefactoring class >> model: aModel find: aMethod of: aClass inAllHierarchy: aBoolean [
+	^ (self new)
+		model: aModel; 
+		find: aMethod 
+		of: aClass
+		inAllHierarchy: aBoolean;
+		yourself
+]
+
+{ #category : #accessing }
+RBFindAndReplaceRefactoring >> argumentsOf: aDictionary [
+	|args limit|
+	limit := self method ast arguments size - 1.
+	args := OrderedCollection new.
+	0 to: limit do: [ :each |
+		args add: 
+			(aDictionary at: (aDictionary keys detect: 
+				[ :e | (e name asString) =  ('`@argMatch', each asString)])) sourceCode
+	 ].
+	^ args
+]
+
+{ #category : #accessing }
+RBFindAndReplaceRefactoring >> copyOptionsFrom: aDictionary [ 
+	| dict |
+	dict := self options.
+	dict == self class refactoringOptions 
+		ifTrue: [^self options: aDictionary copy].
+	dict keysAndValuesDo: 
+			[:key :value | 
+			value == (self class refactoringOptions  at: key ifAbsent: [nil]) 
+				ifTrue: [
+					
+					dict at: key put: (aDictionary at: key)]].
+	(aDictionary keys difference: dict keys) do: 
+		[ :e | dict at: e put: (aDictionary at: e) ].
+	self options: dict
+]
+
+{ #category : #accessing }
+RBFindAndReplaceRefactoring >> extract: ocurrence of: aMethod [
+	[|refactoring |
+	refactoring := self extractMethodRefactoring. 
+	refactoring model: self model.
+	refactoring 	extract: ocurrence key from: aMethod selector in: aMethod modelClass.
+	refactoring setOption: #existingSelector toUse:  [ :ref | 
+			ref parameters: (self argumentsOf: ocurrence value).
+			self method selector].
+	self performComponentRefactoring: refactoring ] on: Exception do: [ :e | e ]
+]
+
+{ #category : #accessing }
+RBFindAndReplaceRefactoring >> extractMethodRefactoring [
+	^ RBExtractMethodRefactoring new
+			setOption: #useExistingMethod 
+			toUse:  [ :ref :aSelector | 
+				true];
+			yourself 
+]
+
+{ #category : #initialization }
+RBFindAndReplaceRefactoring >> find: aSelector of: aClass inAllHierarchy: aBoolean [
+	class := self classObjectFor: aClass.
+	selector := aSelector.
+	replacesAllHierarchy := aBoolean.
+]
+
+{ #category : #accessing }
+RBFindAndReplaceRefactoring >> findAndReplaceOcurrencesIn: aMethod [ 
+	self findOcurrencesIn: aMethod
+]
+
+{ #category : #accessing }
+RBFindAndReplaceRefactoring >> findOcurrencesIn: aMethod [
```

Comment:
findOcurrencesIn -> findOccurrencesIn

Do we have tests for this method because I do not like the onFailure: doing nothing. 
And the method looks a bit too long to my taste